src/lib.rs

```diff
@@ -3293,9 +3308,23 @@ fn which(tool: &Path) -> Option<PathBuf> {
     }
 
     // Loop through PATH entries searching for the |tool|.
-    let path_entries = env::var_os("PATH")?;
+    let path_entries = path_entries.or(env::var_os("PATH"))?;
     env::split_paths(&path_entries).find_map(|path_entry| {
         let mut exe = path_entry.join(tool);
         return if check_exe(&mut exe) { Some(exe) } else { None };
     })
 }
+
+// search for |prog| on 'programs' path in '|cc| -print-search-dirs' output
+fn search_programs(cc: &mut Command, prog: &str) -> Option<PathBuf> {
+    let search_dirs = run_output(cc.arg("-print-search-dirs"), "cc").ok()?;
+    // clang driver appears to be forcing UTF-8 output even on Windows,
+    // hence from_utf8 is assumed to be usable in all cases.
+    let search_dirs = std::str::from_utf8(&search_dirs).ok()?;
+    for dirs in search_dirs.split(|c| c == '\r' || c == '\n') {
+        if dirs.starts_with("programs: =") {
```

Comment:
Can this use `.strip_prefix` to avoid the indexing and hardcoding 11 below?

Reply:
doc.rust-lang.org says it's available in 1.45.0, while your MSRV is 1.34.0. Or do I interpret 1.45.0 at [strip_prefix](https://doc.rust-lang.org/std/string/struct.String.html#method.strip_prefix) incorrectly?